### PR TITLE
logging: default log to a file

### PR DESCRIPTION
**Pull request by ZhouXing19**
Release note: None

### CODE REVIEW

File: cmd/fetch/fetch.go [hunk 1]
```diff
@@ -253,8 +255,8 @@ func Command() *cobra.Command {
 	cmd.PersistentFlags().StringVar(
 		&logFile,
 		"log-file",
-		"",
-		"If set, writes to the log file specified. Otherwise, only writes to stdout.",
+		fmt.Sprintf("fetch-%s.log", time.Now().UTC().Format(time.RFC3339)),
+		"Default write it to a file with format \"fetch-{datetime}.log\". If is set with \"stdout\", write to stdout only",
```
Comment (Jeremyyang920) [inline cmd/fetch/fetch.go:259]: So is the behavior now always print to stdout and log a file by default?

Comment (ZhouXing19) [reply cmd/fetch/fetch.go:259]: yeah, the new default is {stdout + to a file}. Open to revision here

**Comment by ryanluu12345** [comment]
I think this is a broader question, but do we want to add an item to support log rotation? I think the way we do it right now to a single file won't be sustainable especially for **very large migrations**

**Comment by ZhouXing19** [comment]
> I think this is a broader question, but do we want to add an item to support log rotation? I think the way we do it right now to a single file won't be sustainable especially for very large migrations

This is a valid issue. I think we can make it a follow-up. The DB's log is a good example to follow.

File: cmd/fetch/fetch.go (hunk 1 above)
Comment (Jeremyyang920) [reply cmd/fetch/fetch.go:259]: I think that should be fine. Just like ryan pointed out we might have really large log files but that can be a future thing.

```suggestion
		fmt.Sprintf("fetch-%s.log", time.Now().UTC().Format(time.RFC3339)),
		"By default, writes to a file named \"fetch-{datetime}.log\". If set as \"stdout\", writes will only go to stdout",
```
```suggestion
		fmt.Sprintf("fetch-%s.log", time.Now().UTC().Format(time.RFC3339)),
		"Default write it to a file with format \"fetch-{datetime}.log\". If is set with \"stdout\", write to stdout only",
```

Comment (Jeremyyang920) [reply cmd/fetch/fetch.go:259]: Oh and I think we dont need to escape the quotes here since its just output text.

**Comment by ZhouXing19** [comment]
Issue to track log rotation: https://cockroachlabs.atlassian.net/browse/CC-28385

**Comment by ZhouXing19** [comment]
RFAL, thanks!

File: fetch/fetch.go [hunk 2]
```diff
@@ -479,13 +477,17 @@ func Fetch(
...
+		i := i
 		g.Go(func() error {
 			for {
 				table, ok := <-workCh
 				if !ok {
 					return nil
 				}
 
+				loggerWithConcurrencyIdx := logger.With().Str("table-concurrency-idx", fmt.Sprintf("%d", utils.MaybeFormatIndex(cfg.TestOnly, i))).Logger()
+				loggerWithConcurrencyIdx.Info().Msgf("start fetching table %s", table.SafeString())
```
Comment (Jeremyyang920) [inline fetch/fetch.go:489]:
```suggestion
				loggerWithConcurrencyIdx.Info().Msgf("starting fetch for table %s", table.SafeString())
```

File: moltlogger/logger.go [hunk 3]
```diff
@@ -49,15 +50,18 @@ func RegisterLoggerFlags(cmd *cobra.Command) {
...
+func Logger(fileName string, testOnly bool) (zerolog.Logger, error) {
 	var writer io.Writer = os.Stdout
 	if loggerConfigInst.useConsoleWriter {
 		writer = zerolog.NewConsoleWriter(func(w *zerolog.ConsoleWriter) {
 			w.TimeFormat = time.RFC3339
 		})
 	}
 
-	if fileName != "" {
+	if fileName != "stdout" && !testOnly {
```
Comment (Jeremyyang920) [inline moltlogger/logger.go:61]: We should use strings.equalfolds here for case insensitivity.

**Review by Jeremyyang920 — APPROVED** [review]
LGTM with just the 2 small comments.

File: moltlogger/logger.go [hunk 4]
```diff
@@ -49,15 +51,18 @@ func RegisterLoggerFlags(cmd *cobra.Command) {
...
+func Logger(fileName string, testOnly bool) (zerolog.Logger, error) {
 	var writer io.Writer = os.Stdout
 	if loggerConfigInst.useConsoleWriter {
 		writer = zerolog.NewConsoleWriter(func(w *zerolog.ConsoleWriter) {
 			w.TimeFormat = time.RFC3339
 		})
 	}
 
-	if fileName != "" {
+	if strings.EqualFold(fileName, "stdout") && !testOnly {
```
Comment (ryanluu12345) [inline moltlogger/logger.go:62]: @ZhouXing19 I just noticed this bug while testing on the DRT cluster. Previously, this `fileName` check intends to create a multi writer in the case that there is a file name specified by the user. However, with the logic change here, we don't do anything if the user passes in a file name. This renders the `--log-file` override flag useless. 

I think the correct behavior is to:
1. If stdout is default, then create the file name for them
2. If stdout is not default, and not empty, then just use that filename

So what this logic ends up changing to is:
```
if !testOnly {
   if fileName == "stdout" {
			fileName = fmt.Sprintf("fetch-%s.log", time.Now().UTC().Format(time.RFC3339))
   }
}
```

File: moltlogger/logger.go [hunk 5]
```diff
@@ -49,15 +51,18 @@ func RegisterLoggerFlags(cmd *cobra.Command) {
...
+		if fileName == "" {
```
Comment (ryanluu12345) [inline moltlogger/logger.go:63]: Also if I'm not mistaken, this code block is unreachable right? The reason I'm saying this is: we only enter this block if fileName == "stdout". So there is no way for `fileName` to be empty haha.

Comment (ryanluu12345) [reply moltlogger/logger.go:63]: I can take care of this fix

File: moltlogger/logger.go (hunk 4 above)
Comment (ryanluu12345) [reply moltlogger/logger.go:62]: Nevermind, it's even simpler. The fix that we need here is this:
```
if !strings.EqualFold(fileName, "stdout") && !testOnly {
```

Comment (ryanluu12345) [reply moltlogger/logger.go:62]: Fixed here: https://github.com/cockroachdb/molt/pull/257

**Pull request merged**